ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/MessageListContainerFragmentTest.java

```diff
@@ -0,0 +1,403 @@
+
+    // Act
+    messageListContainerFragment.onDestroyView();
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).onDestroyView();
+  }
+
+  @Test
+  public void testHasPageLoaded() {
+    // Arrange
+    boolean isAdded = messageListContainerFragment.isAdded();
+    Activity activity = messageListContainerFragment.getActivity();
+
+    // Act
+    boolean pageLoaded = messageListContainerFragment.hasPageLoaded();
+
+    // Assert
+    Mockito.verify(messageListContainerFragment, Mockito.times(2)).isAdded();
+    if (isAdded) {
+      Mockito.verify(messageListContainerFragment, Mockito.times(2)).getActivity();
+    }
+    assertThat(pageLoaded, is(isAdded && activity != null));
+  }
+
+  @Test
+  public void testShowToastMessageWithResId() throws Exception {
+    // Arrange
+    Toast toastMock = Mockito.mock(Toast.class);
+    PowerMockito.when(Toast.class, "makeText", context, R.string.ko__action_contact, Toast.LENGTH_SHORT)
+        .thenReturn(toastMock);
+
+    // Act
+    messageListContainerFragment.showToastMessage(R.string.ko__action_contact);
+
+    // Assert
+    Mockito.verify(toastMock).show();
+  }
+
+  @Test
+  public void testShowToastMessageWithString() throws Exception {
+    // Arrange
+    Toast toastMock = Mockito.mock(Toast.class);
+    PowerMockito.when(Toast.class, "makeText", context, TOAST_MESSAGE, Toast.LENGTH_SHORT)
+        .thenReturn(toastMock);
+
+    // Act
+    messageListContainerFragment.showToastMessage(TOAST_MESSAGE);
+
+    // Assert
+    Mockito.verify(toastMock).show();
+  }
+
+  @Test
+  public void testSetupListInMessageListingView() {
+    // Arrange
+    List<BaseListItem> baseListItems = new ArrayList<>();
+    PowerMockito.when(messageListContainerFragment.hasPageLoaded()).thenReturn(true);
+    MessageListContract.ConfigureView messageListViewMock = Mockito.mock(MessageListContract.ConfigureView.class);
+    MessageListContainerContract.Presenter presenterMock = Mockito.mock(MessageListContainerContract.Presenter.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mMessageListView", messageListViewMock);
+    Whitebox.setInternalState(messageListContainerFragment, "mPresenter", presenterMock);
+
+    // Act
+    messageListContainerFragment.setupListInMessageListingView(baseListItems);
+
+    // Assert
+    Mockito.verify(messageListContainerFragment).setupListInMessageListingView(baseListItems);
+    Mockito.verify(messageListViewMock).setupList(baseListItems);
+  }
+
+  @Test
+  public void testShowEmptyViewInMessageListingView() {
+    // Arrange
+    PowerMockito.when(messageListContainerFragment.hasPageLoaded()).thenReturn(true);
+    MessageListContract.ConfigureView messageListViewMock = Mockito.mock(MessageListContract.ConfigureView.class);
+    Whitebox.setInternalState(messageListContainerFragment, "mMessageListView", messageListViewMock);
+
+    // Act
+    messageListContainerFragment.showEmptyViewInMessageListingView();
+
+    // Assert
```

Comment:
✅ Good to see that the AAA("Arrange-Act-Assert") arranging and formatting pattern is followed.

Some of the main advantages are that it clearly separates what is being tested from the setup and verification steps. It also clarifies and focuses attention on a historically successful and generally necessary set of test steps.

**No change suggested, looks good!**	  

<a id="99938d30-4f1d-11e8-b712-a55fdd633c3d" href="none" codereview-uuid-reserved-tag />